### PR TITLE
bugfix/invoice-transaction-id-adjustment

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
## PR Overview
**This PR will address the following Issue/Feature:** Internally created issue

**This PR will result in the following new package version:** `v0.15.0`

While not a traditional breaking change, we made this a minor upgrade to account for scenarios where the end model results will likely change due to invoices being attributed to the correct accounts.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## Bug Fixes
> - Updates the `int_quickbooks__sales_receipt_double_entry` model to prioritize the `invoice_lines.sales_item_account_id` as the second viable option in the `account_id` coalesce statements. This field was previously prioritized last. However, recent observations have made it apparent that when prioritized last, invoice transactions could be attributed to the wrong accounts.
>   - While not a traditional breaking change, we made this a minor upgrade to account for scenarios where the end model results will likely change due to invoices being attributed to the correct accounts.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [n/a] dbt run –full-refresh && dbt test
- [X] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [X] The appropriate issue has been linked, tagged, and properly assigned
- [X] All necessary documentation and version upgrades have been applied
- [X] docs were regenerated (unless this PR does not include any code or yml updates)
- [X] BuildKite integration tests are passing
- [X] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

image

### If you had to summarize this PR in an emoji, which would it be?
 
🥗

### CODE REVIEW

**Review by fivetran-catfritz — APPROVED** [review]
@fivetran-joemarkiewicz I gave this a review, run, and test, and all looks good to me!

File: models/double_entry_transactions/int_quickbooks__invoice_double_entry.sql [hunk 1]
```diff
@@ -112,21 +112,21 @@ invoice_join as (
...
             when invoice_lines.discount_account_id is not null then 'DiscountLineDetail'
-            when coalesce(invoice_lines.account_id, items.parent_income_account_id, items.income_account_id, bundle_income_accounts.account_id, invoice_lines.discount_account_id, invoice_lines.sales_item_account_id) is null then 'NoAccountMapping'
+            when coalesce(invoice_lines.account_id, invoice_lines.sales_item_account_id, items.parent_income_account_id, items.income_account_id, bundle_income_accounts.account_id, invoice_lines.discount_account_id) is null then 'NoAccountMapping'
         end as invoice_line_transaction_type,
-        coalesce(invoice_lines.account_id, items.parent_income_account_id, items.income_account_id, bundle_income_accounts.account_id, invoice_lines.discount_account_id, invoice_lines.sales_item_account_id) as account_id,
+        coalesce(invoice_lines.account_id, items.parent_income_account_id, items.income_account_id, bundle_income_accounts.account_id, invoice_lines.discount_account_id) as account_id,
```
Comment (fivetran-reneeli) [inline models/double_entry_transactions/int_quickbooks__invoice_double_entry.sql:119]: I noticed elsewhere, the `sales_item_account_id` got moved to 2nd in order, but in line 119, it got deleted-- could you clue me into was the reason for this was?

Comment (fivetran-joemarkiewicz) [reply models/double_entry_transactions/int_quickbooks__invoice_double_entry.sql]: @fivetran-reneeli fantastic catch! You're right this should not have been removed and I missed it in my copy/pasting.

Update applied and docs regenerated.

**Review by fivetran-reneeli — APPROVED** [review]
lgtm!

**Pull request merged**